test/unittests/CandidateTest.php

```diff
@@ -11,6 +11,71 @@
  * @link     https://www.github.com/aces/Loris/
  */
 use PHPUnit\Framework\TestCase;
+
+/**
+ * Fake Candidate class that has exact copies of methods in the
+ * Candidate class to test certain SQL queries easily
```

Comment:
Can you change the code so calling `$DB           =& Database::singleton();` would use the fake DB? i.e. using different access parametre than in the config.xml file to connect to the DB?

This would prevent you code from being obsolete in case the actual libraries get updated/modified